catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/cache/impl/CacheKey.java

```diff
@@ -36,15 +37,16 @@ public CacheKey(Metacard metacard, ResourceRequest resourceRequest) {
      * {@code <sourceId>-<metacardId>[-<RESOURCE_OPTION>]}
      *
      * @return key
+     * @throws CacheException if the metatcard or resourceRequest are null
      */
-    public String generateKey() {
+    public String generateKey() throws CacheException {
 
         if (metacard == null) {
-            throw new IllegalArgumentException("Metacard must not be null.");
+            throw new CacheException("Metacard must not be null.");
         }
 
         if (resourceRequest == null) {
-            throw new IllegalArgumentException("ResourceRequest must not be null.");
+            throw new CacheException("ResourceRequest must not be null.");
         }
 
```

Comment:
If the client code hasn't initialized this object correctly before calling this method, then they've violated the contract of the class.  There are a few ways to prevent this, but throwing a custom checked exception would be my last choice.  If we're going to throw an exception, it should probably be IllegalStateException.